src/live/handleAlbumChat.js

```diff
@@ -1,11 +1,36 @@
-function handleAlbumChat(io, socket, data) {
-  const { message, userId, albumId } = data;
+const ChatMessage = require('../models/ChatMessage');
 
-  // Create a chat room name for the album chat
-  const chatRoomName = `chat:album:${albumId}`;
+async function handleAlbumChat(io, socket, data) {
+  try {
+    const { albumId } = data;
 
-  // Send the message to the room
-  socket.to(chatRoomName).emit('chat:album', { message, userId, albumId });
+    // Create a chat room name for the album chat
+    const chatRoomName = `chat:album:${albumId}`;
+
+    const msg = await ChatMessage.create({ ...data, name: data.user });
+    await msg.populate('user', 'name');
+    console.log('Chat message saved:', msg);
+
+    const savedMessage = {
+      ...data,
+      _id: msg._id, // Include _id in the message
+      name: data.name,
```

Comment:
The  `data` object sent to us by the frontend does not have a `name`  field, so this will always be undefined: https://github.com/Code-the-Dream-School/dd-prac-team4-front/pull/52/files#diff-dc044c6ce14501135532d751cb2c59b15079b67646f1634abe6dd9c739c899b4R26

Reply:
I have already merged the frontend. This means I still need to change the frontend. I really can't understand the logic. As I understand it, I don’t need to add a name to the message scheme, since the name data is already in the user schema. I created a new user with Name and Username. But I don’t understand the logic at what point I should enter the name for it to work. I will try think about this tomorrow more, im sorry😒

Reply:
:wave: I don't believe you'll need to change the frontend. You're correct that name is in the user schema of the message but you're returning a mixed object rather than the `ChatMessage` itself here. Recorded a video I think I'll be able to show what I mean better that way, no need to apologize, sometimes over text things get mixed up or are not as clear! Let me know if you have any questions:

https://drive.google.com/file/d/1hONQ_OrdKTAytdUvjyVdHyv2PtzG5naT/view?usp=sharing

Reply:
@akosasante 🤦‍♀️i dont know what to say😅... I apologize for my confusion, it was such straightforward logic. Thank you so much for the explanation.